src/Sylius/Component/Core/Model/OrderItemUnit.php

```diff
@@ -0,0 +1,200 @@
+
+use Sylius\Component\Inventory\Model\InventoryUnitInterface;
+use Sylius\Component\Inventory\Model\StockableInterface;
+use Sylius\Component\Order\Model\OrderItemUnit as BaseOrderItemUnit;
+use Sylius\Component\Shipping\Model\ShipmentInterface as BaseShipmentInterface;
+use Sylius\Component\Shipping\Model\ShippableInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class OrderItemUnit extends BaseOrderItemUnit implements OrderItemUnitInterface
+{
+    /**
+     * @var string InventoryUnitInterface::STATE_*
+     */
+    protected $inventoryState = InventoryUnitInterface::STATE_CHECKOUT;
+
+    /**
+     * @var BaseShipmentInterface
+     */
+    protected $shipment;
+
+    /**
+     * @var string BaseShipmentInterface::STATE_*
+     */
+    protected $shippingState = BaseShipmentInterface::STATE_CHECKOUT;
+
+    /**
+     * @var \DateTime
+     */
+    protected $createdAt;
+
+    /**
+     * @var \DateTime
+     */
+    protected $updatedAt;
+
+    public function __construct(OrderItemInterface $orderItem)
+    {
+        parent::__construct($orderItem);
+
+        $this->createdAt = new \DateTime();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getStockable()
+    {
+        return $this->orderItem->getVariant();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setStockable(StockableInterface $stockable)
+    {
+        $this->orderItem->setVariant($stockable);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSku()
+    {
+        return $this->orderItem->getVariant()->getSku();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInventoryName()
+    {
+        return $this->orderItem->getVariant()->getInventoryName();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getInventoryState()
+    {
+        return $this->inventoryState;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setInventoryState($state)
+    {
+        $this->inventoryState = $state;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSold()
+    {
+        return InventoryUnitInterface::STATE_SOLD === $this->inventoryState;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isBackordered()
+    {
+        return InventoryUnitInterface::STATE_BACKORDERED === $this->inventoryState;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getShipment()
+    {
+        return $this->shipment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setShipment(BaseShipmentInterface $shipment = null)
+    {
+        $this->shipment = $shipment;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getShippable()
+    {
+        return $this->orderItem->getVariant();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setShippable(ShippableInterface $shippable)
+    {
+        $this->orderItem->setVariant($shippable);
```

Comment:
I don't think this method is necessary nor safe to use because the variant shouldn't change on order item. @pjedrzejewski do we use it at all?